### PR TITLE
Voice memos

### PR DESCRIPTION
**Pull request by daria-github**
Added voice recording functionality through a custom hook and via `react-media-recorder`, which asks permission for mic on mount, then enables recording. 

While recording, displays a stopwatch with the recording length with an obvious red banner. Records on hold. Also, made some accessibility tweaks here so that the recording can also happen via keyboard, and some accessibility fixes to the other svgs that weren't yet keyboard-enabled. 

Updated to a smaller package to import the attachment types as well. 

**Via mouse (holding down):**

https://github.com/xmtp-labs/xmtp-inbox-web/assets/35409260/6df90eca-52e7-42f8-9ca3-2189163b48ee

**Via keyboard:**

https://github.com/xmtp-labs/xmtp-inbox-web/assets/35409260/1866a1ad-1687-49eb-a00a-617769aee27f

### CODE REVIEW

**Review by nakajima — APPROVED** [review]

File: src/component-library/components/MessageInput/MessageInput.tsx [hunk 1]
```diff
@@ -251,6 +266,21 @@ export const MessageInput = ({
...
+              onClick={stopRecording}
+            />
+          ) : (
+            <MicrophoneIcon
+              width={24}
+              height={24}
+              className="m-2 cursor-pointer text-gray-400 hover:text-black"
+              onClick={startRecording}
+            />
+          )}
```
[the remark this answers is not in the document]
Comment (daria-github) [reply src/component-library/components/MessageInput/MessageInput.tsx:283]: great suggestion. thanks.

File: src/helpers/recordingValue.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * Example:
+ * ```
+ * const value = getRecordingValue('recording', 5, 7, 'Recording');
+ * // value = 'Recording (05:07)'
+ * ```
+ */
+
+export const getRecordingValue = (
+  status: string,
```
Comment (rygine) [inline src/helpers/recordingValue.ts:20]: should we use a string union here to prevent invalid values?

File: src/hooks/useRecordingTimer.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    const handleKeyDown = () => {
+      startRecording();
+      start();
+    };
+
+    const handleKeyUp = () => {
+      stopRecording();
+      pause();
+      reset();
+    };
```
Comment (rygine) [inline src/hooks/useRecordingTimer.tsx:33]: `useCallback` on these?

File: src/locales/en_US.json [hunk 4]
```diff
@@ -45,7 +45,8 @@
   "attachments": {
     "unable_to_display": "Unable to display attachment.",
     "download_instead": "Download instead.",
-    "video_messages_not_supported": "Your browser does not support the video tag."
+    "video_messages_not_supported": "Your browser does not support the video tag.",
+    "recording": "Recording..."
```
Comment (nakajima) [inline src/locales/en_US.json:49]:
```suggestion
    "recording": "Recording…"
```

**Review by nakajima — APPROVED** [review]

**Pull request merged**